include/RED4ext/NativeTypes.hpp

```diff
@@ -209,18 +212,55 @@ static_assert(sizeof(StaticArray<std::array<uint8_t, 5>, 32>) ==
 template<typename T, uint32_t MAX_LEN>
 using NativeArray = std::array<T, MAX_LEN>;
 
+template<typename T>
+struct CurvePoint
+{
+    float point;
+    T value;
+};
+
+template<typename T>
+struct CurveBuffer
+{
+    float* GetPoints() const noexcept;
+    T* GetValues() const noexcept;
```

Comment:
```suggestion
    [[nodiscard]] float* GetPoints() const noexcept;
    [[nodiscard]] T* GetValues() const noexcept;
```

Reply:
I'll add `[[nodiscard]]` then, I wasn't sure as it was not present in other parts of the code I looked at.